xarray/core/dataset.py

```diff
@@ -1904,19 +1915,21 @@ def isel(
                 )
                 if new_index is not None:
                     indexes[name] = new_index
-            else:
+            elif var_indexers:
                 new_var = var.isel(indexers=var_indexers)
+            else:
+                new_var = var
```

Comment:
Can we make this:
```suggestion
                new_var = var.copy(deep=False)
```
Otherwise, this would mean that you could modify attributes on variables on the resulting dataset and change the input dataset.

Reply:
Fixed. Good catch!